src/lib/docs/is-release-notes-page.ts

```diff
@@ -0,0 +1,22 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+/**
+ * Determines if a given path corresponds to a release notes page.
+ *
+ * This function uses a regular expression to check if the provided path matches
+ * the expected patterns for release notes pages. The patterns include:
+ * - `vYYYYMM-NN/releases/YYYY/vYYYYMM-NN`
+ * - `releases/YYYY/vYYYYMM-NN`
+ * - `/releases/YYYY/vYYYYMM-NN`
+ * - `release-notes/vX.X.X` or `/release-notes/X.X.X`
+ *
+ * @param path - The path to be checked.
+ * @returns `true` if the path matches the release notes pattern, otherwise `false`.
+ */
+export const isReleaseNotesPage = (path: string) =>
+	/(\/?releases\/\d{4}\/(v\d{6}-\d{1}))$|\/?release-notes\/(v\d+[.|_]|(\d+[.|_]))\d+[.|_]([0-9]|x)$/i.test(
+		path
```

Comment:
Regex can be a challenge for humans to parse. There could be an advantage here to creating an array of Regex conditions for each of the 4 patterns, and then using array.some() to test for a match.


```
const  patterns = [ /regex-pattern1/, /regex-pattern-2/]

return patterns.some(pattern => pattern.test(path))

```

Reply:
Do you think having the examples in the function comment would help with this?
```
 * - `vYYYYMM-NN/releases/YYYY/vYYYYMM-NN`
 * - `releases/YYYY/vYYYYMM-NN`
 * - `/releases/YYYY/vYYYYMM-NN`
 * - `release-notes/vX.X.X` or `/release-notes/X.X.X`
 ```

Reply:
done in https://github.com/hashicorp/dev-portal/pull/2576/commits/2a22362c9e179b1e8d50ac53dbe05185bca28e9e